File: backend/goat_listing/src/main/java/com/goat/cart/domain/exceptions/CannotAddOwnListingException.java
package com.goat.cart.domain.exceptions;

/**
 * Excepción lanzada cuando un usuario intenta agregar su propio listing al carrito.
 */
public class CannotAddOwnListingException extends RuntimeException {
    public CannotAddOwnListingException(String message) {
        super(message);
    }

    public CannotAddOwnListingException(String message, Throwable cause) {
        super(message, cause);
    }
}

File: backend/goat_catalog/src/goat_catalog/cart/domain/exceptions/cannot_add_own_listing_exception.py
"""Excepción cuando se intenta agregar un listing propio."""


class CannotAddOwnListingException(Exception):
    """Excepción lanzada cuando un seller intenta agregar su propio listing."""

    def __init__(self, message: str = "No puedes agregar tus propios listings al carrito") -> None:
        """Inicializa la excepción.

        Args:
            message: Mensaje de error
        """
        super().__init__(message)
        self.message = message

